django/contrib/gis/gdal/raster/source.py

```diff
@@ -103,33 +186,71 @@ def srs(self):
         Returns the Spatial Reference used in this GDALRaster.
         """
         try:
-            wkt = capi.get_ds_projection_ref(self.ptr)
+            wkt = capi.get_ds_projection_ref(self._ptr)
+            if not wkt:
+                return None
             return SpatialReference(wkt, srs_type='wkt')
         except SRSException:
             return None
 
-    @cached_property
+    @srs.setter
+    def srs(self, value):
+        """
+        Sets the Spatial Reference used in this GDALRaster. The input can be
+        a SpatialReference or any parameter accepted by the SpatialReference
+        constructor.
+        """
+        if isinstance(value, SpatialReference):
+            srs = value
+        elif isinstance(value, six.integer_types + six.string_types):
+            srs = SpatialReference(value)
+        else:
+            raise ValueError('Could not create a SpatialReference from input.')
+        capi.set_ds_projection_ref(self._ptr, srs.wkt.encode())
+        self._flush()
+
+    @property
     def geotransform(self):
         """
         Returns the geotransform of the data source.
         Returns the default geotransform if it does not exist or has not been
-        set previously. The default is (0.0, 1.0, 0.0, 0.0, 0.0, -1.0).
+        set previously. The default is [0.0, 1.0, 0.0, 0.0, 0.0, -1.0].
         """
         # Create empty ctypes double array for data
         gtf = (c_double * 6)()
-        capi.get_ds_geotransform(self.ptr, byref(gtf))
-        return tuple(gtf)
+        capi.get_ds_geotransform(self._ptr, byref(gtf))
+        return list(gtf)
+
+    @geotransform.setter
+    def geotransform(self, value):
+        "Sets the geotransform for the data source."
+        if sum([isinstance(x, (int, float)) for x in value]) != 6:
+            raise ValueError('Geotransform must consist of 6 numeric values.')
+        # Create ctypes double array with input and write data
+        value = (c_double * 6)(value[0], value[1], value[2],
```

Comment:
We have a way of using names to change the geotransform using the `origin`, `scale` and `skew` attributes through the `TransformPoint` objects. I think if we want to name the values here, we would have to change that logic too, as it relies on the array and the corresponding positions. Currently the geotransform array is a property. But you might be right, naming them would be much easier to use when working with the geotransform attribute. I am open to both, what do you think?